bitnami/metallb/templates/deployment.yaml

```diff
@@ -0,0 +1,68 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "metallb.fullname" . }}-controller
+  labels: {{- include "metallb.labels" . | nindent 4 }}
+    component: controller
+spec:
+  revisionHistoryLimit: 3
+  selector:
+    matchLabels: {{- include "metallb.matchLabels" . | nindent 6 }}
+      component: controller
+  template:
+    metadata:
+      labels: {{- include "metallb.labels" . | nindent 8 }}
+        component: controller
+      {{- if .Values.controller.podAnnotations }}
+      annotations: {{- toYaml .Values.controller.podAnnotations | nindent 8 }}
+      {{- end }}
+    spec:
+{{- include "metallb.controller.imagePullSecrets" . | nindent 6 }}
+      serviceAccountName: {{ include "metallb.controllerServiceAccountName" . }}
+      terminationGracePeriodSeconds: 0
+      securityContext:
+        runAsNonRoot: true
+        runAsUser: 65534 # nobody
+      nodeSelector:
+        "beta.kubernetes.io/os": linux
+        {{- with .Values.controller.nodeSelector }}
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.controller.tolerations }}
+      tolerations:
+{{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- with .Values.controller.affinity }}
+      affinity:
+{{ toYaml . | indent 8 }}
```

Comment:
We usually use a specific function to import this kind of settings. For example: https://github.com/bitnami/charts/blob/de6dba952b698c6e94b8caa8950e947275d700dc/bitnami/grafana/templates/deployment.yaml#L34

The `tplValue` is a function that renders what is in the `values.yaml` before set it in the resource. In that way you can use templates in the `values.yaml`:
```
{{- define "grafana.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

Reply:
know that dont't know that templating is used for affinity settings ^^